.github/workflows/win.yml

```diff
@@ -0,0 +1,134 @@
+        name: Bootstrap vcpkg and install packages (if cache miss)
+        run: |
+          git clone --depth=1 --branch "$VCPKG_VERSION" https://github.com/microsoft/vcpkg.git
+          cd vcpkg
+          mkdir archives
+          mkdir "$VCPKG_OVERLAY_TRIPLETS"
+          TRIPLET_FILE="$VCPKG_OVERLAY_TRIPLETS/$VCPKG_DEFAULT_TRIPLET.cmake"
+          cp triplets/x64-windows.cmake "$TRIPLET_FILE"
+          echo "set(VCPKG_BUILD_TYPE release)" >> "$TRIPLET_FILE"
+          echo "set(VCPKG_DISABLE_COMPILER_TRACKING $VCPKG_DISABLE_COMPILER_TRACKING)" >> "$TRIPLET_FILE"
+          cmd.exe /c bootstrap-vcpkg.bat
+          vcpkg.exe \
+            --vcpkg-root="$VCPKG_ROOT" \
+            --overlay-triplets="$VCPKG_OVERLAY_TRIPLETS" \
+            --triplet="$VCPKG_DEFAULT_TRIPLET" \
+            $VCPKG_INSTALL_OPTIONS \
+            install
+
+      - name: Setup Developer Command Prompt for VS
+        uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Configure CMake
+        run: |
+          cmake --version
+          cmake -B build \
+            -G "Visual Studio 17 2022" \
+            -A x64 \
+            -DCMAKE_BUILD_TYPE="$BUILD_TYPE" \
+            -DCMAKE_TOOLCHAIN_FILE="$VCPKG_TOOLCHAIN_FILE" \
+            -DVCPKG_OVERLAY_TRIPLETS="$VCPKG_OVERLAY_TRIPLETS" \
+            -DVCPKG_TARGET_TRIPLET="$VCPKG_DEFAULT_TRIPLET" \
+            -DVCPKG_INSTALL_OPTIONS="$VCPKG_INSTALL_OPTIONS" \
+            -DENABLE_DATA_TOOLS=ON \
+            -DENABLE_TOOLS=ON \
+            -DENABLE_PYTHON_BINDINGS=ON \
+            -DENABLE_TESTS=OFF \
+            -DENABLE_CCACHE=OFF \
+            -DENABLE_HTTP=OFF \
+            -DENABLE_SERVICES=OFF \
+            -DENABLE_BENCHMARKS=OFF
+
+      - name: Build Valhalla
+        run: |
+          cmake --build build --config $BUILD_TYPE -- /clp:ErrorsOnly
+      
+      - name: Test Executable
+        shell: cmd
+        run: |
+          set PATH=$PATH;${{ github.workspace }}\build\vcpkg_installed\$VCPKG_DEFAULT_TRIPLET\bin
+          ${{ github.workspace }}\build\$BUILD_TYPE\valhalla_build_tiles.exe -c .\test\win\valhalla.json .\test\data\utrecht_netherlands.osm.pbf
+          ${{ github.workspace }}\build\$BUILD_TYPE\valhalla_run_route.exe --config .\test\win\valhalla.json -j "{\"locations\": [{\"lat\": 52.10205, \"lon\": 5.114651}, {\"lat\": 52.093113, \"lon\": 5.100918}], \"costing\": \"auto\"}"
+          ${{ github.workspace }}\build\$BUILD_TYPE\valhalla_run_isochrone.exe --config .\test\win\valhalla.json -j "{\"locations\": [{\"lat\": 52.10205, \"lon\": 5.114651}], \"costing\": \"auto\", \"contours\":[{\"time\":15,\"color\":\"ff0000\"}]}"
+      
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3
+        # only run this if manually invoked or a previous job failed
+        if: ${{ (github.event_name == 'workflow_dispatch' && inputs.debug_enabled) || failure() }}
+        with:
+          detached: true
+          timeout-minutes: 15
```

Comment:
super useful action, opens a SSH session on the github server if:
- we triggered it manually and checked the boolean flag when doing so, in case we'd like to try some stuff on windows
- there was a failure in any step in which case it'll wait for 15 minutes before terminating the SSH session